lib/auth/v2/checkRequestExpiry.js

```diff
@@ -1,24 +1,36 @@
 'use strict'; // eslint-disable-line strict
+const errors = require('../../errors');
 
 function checkRequestExpiry(timestamp, log) {
+    const epochTime = new Date('1970-01-01').getTime();
+    // If timestamp is before epochTime, the request is invalid and return
+    // errors.AccessDenied
+    if (timestamp < epochTime) {
+        log.trace('request timestamp is before epochTime');
+        log.debug('request time is invalid', { timestamp });
+        return errors.AccessDenied;
+    }
     // If timestamp is not within 15 minutes of current time, or if
     // timestamp is more than 15 minutes in the future, the request
-    // has expired and return true
+    // has expired and return errors.RequestTimeTooSkewed
     const currentTime = Date.now();
     log.trace('request timestamp', { requestTimestamp: timestamp });
     log.trace('current timestamp', { currentTimestamp: currentTime });
+
     const fifteenMinutes = (15 * 60 * 1000);
     if (currentTime - timestamp > fifteenMinutes) {
         log.trace('request timestamp is not within 15 minutes of current time');
-        return true;
+        log.debug('request time too skewed', { timestamp });
+        return errors.RequestTimeTooSkewed;
     }
 
     if (currentTime + fifteenMinutes < timestamp) {
         log.trace('request timestamp is more than 15 minutes into future');
-        return true;
+        log.debug('request time too skewed', { timestamp });
+        return errors.RequestTimeTooSkewed;
     }
 
-    return false;
+    return {};
```

Comment:
Should we not use "undefined" here, to imply that there's no error, rather than an empty object ?
this looks like counter-intuitive way, that may go against usual "error" management expectations, and lead to errors where the empty object is seen as an error that does not actually say anything,